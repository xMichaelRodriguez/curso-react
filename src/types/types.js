export const types = {
  //ui
  uiOpenModal: "[UI] Open modal",
  uiCloseModal: "[UI] Close modal",

  //event
  eventSetActive: "[EVENT] Set Active",
  eventAddNew: "[EVENT] Add new",
  eventClearActiveEvent: "[EVENT] Clear active event",
  eventUpdated: "[EVENT] Event updated",
  eventDeleted: "[EVENT] Event deleted",

  //auth

  authFinishChecking: "[AUTH] Finish checking login state",
  authStartLogin: "[AUTH] Start login",
  authLogin: "[AUTH] logged",
  authStartregister: "[AUTH] Start register",
  authStartTokenRenew: "[AUTH] Start token renew",
  authLogout: "[AUTH] Auth logout",
};
