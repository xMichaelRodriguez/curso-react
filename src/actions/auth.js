import Swal from "sweetalert2";
import { fetchAsync, fetchSync } from "../helpers/fetch";
import { types } from "../types/types";
import { eventLogout } from "./events";

export const startLogin = (email, password) => {
  return async (dispatch) => {
    Swal.fire({
      title: "Iniciando Sesion...",
      text: "Por favor espere...",
      allowOutsideClick: false,
      allowEnterKey: false,
      allowEscapeKey: false,
      showConfirmButton: false,
      onBeforeOpen: () => {
        Swal.showLoading();
      },
    });
    const resp = await fetchSync("auth", { email, password }, "POST");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-initDate", new Date().getTime());
      dispatch(login({ uid: body.uid, name: body.name }));
      Swal.close();
    } else {
      Swal.close();
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    Swal.fire({
      title: "Registrando...",
      text: "Por favor espere...",
      allowOutsideClick: false,
      allowEnterKey: false,
      allowEscapeKey: false,
      showConfirmButton: false,
      onBeforeOpen: () => {
        Swal.showLoading();
      },
    });
    const resp = await fetchSync("auth/new", { name, email, password }, "POST");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-initDate", new Date().getTime());
      dispatch(login({ uid: body.uid, name: body.name }));
      Swal.close();
    } else {
      Swal.close();
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startChecking = () => {
  return async (dispatch) => {
    const resp = await fetchAsync("auth/renew");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);

      localStorage.setItem("token-initDate", new Date().getTime());

      dispatch(login({ uid: body.uid, name: body.name }));

      Swal.close();
    } else {
      dispatch(checkingFish());
    }
  };
};

const checkingFish = () => ({
  type: types.authFinishChecking,
});

const login = (user) => ({
  type: types.authLogin,
  payload: user,
});

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();

    dispatch(eventLogout());

    dispatch(logout());
  };
};

const logout = () => ({
  type: types.authLogout,
});
