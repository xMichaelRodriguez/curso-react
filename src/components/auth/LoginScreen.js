import React from "react";
import { FaSignInAlt } from "react-icons/fa";
import { Link } from "react-router-dom";
export const LoginScreen = () => {
  const handleLogin = (e) => e.preventDefault();
  return (
    <div className="shadow auth">
      <form onSubmit={handleLogin}>
        <h1 className="mb-4">
          <ins>Blog</ins>
        </h1>

        <div className="form-row">
          <div className="col-md-12 mb-3">
            <div className="form-group">
              <label>Email</label>
              <input
                className="form-control"
                type="email"
                placeholder="example@dominio.com"
              />
            </div>
          </div>
          <div className="col-md-12 mb-3">
            <label>Password</label>
            <input
              type="password"
              className="form-control"
              placeholder="Password"
            />
          </div>
          <div className="col-md-6 mb-3">
            <button className="btn primary btn-lg" type="submit">
              <FaSignInAlt /> &nbsp;
              <span>Login</span>
            </button>
          </div>
          <div className="col-md-6 w-100">
            <Link to="/auth/register" class="btn btn-link text-capitalize">
              forgot password?
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};
