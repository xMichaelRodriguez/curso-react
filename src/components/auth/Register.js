import React from 'react';
import { FaSignInAlt } from 'react-icons/fa';
import { Link } from 'react-router-dom';

export const Register = () => {
  const handleRegister = (e) => e.preventDefault();
  return (
    <div className='shadow auth animate__animated animate__fadeIn'>
      <form onSubmit={handleRegister}>
        <h1 className='mb-4'>
          <ins>Blog</ins>
        </h1>

        <div className='form-row'>
          <div className='col-md-6'>
            <div className='form-group'>
              <label>First Name</label>
              <input
                className='form-control'
                type='text'
                placeholder='First Name'
              />
            </div>
          </div>
          <div className='col-md-6'>
            <div className='form-group'>
              <label>Last Name</label>
              <input
                className='form-control'
                type='text'
                placeholder='Last Name'
              />
            </div>
          </div>
          <div className='col-md-12 mb-3'>
            <div className='form-group'>
              <label>Email</label>
              <input
                className='form-control'
                type='email'
                placeholder='example@dominio.com'
              />
            </div>
          </div>
          <div className='col-md-12 mb-3'>
            <label>Password</label>
            <input
              type='password'
              className='form-control'
              placeholder='Password'
            />
          </div>
          <div className='col-md-6 mb-3'>
            <button className='btn primary btn-lg' type='submit'>
              <FaSignInAlt /> &nbsp;
              <span>Register</span>
            </button>
          </div>
          <div className='col-md-6 w-100'>
            <Link to='/auth/' class='btn btn-link text-capitalize'>
              I have an account
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};
