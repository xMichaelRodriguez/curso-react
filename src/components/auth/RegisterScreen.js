import React from "react";
import { Link } from "react-router-dom";

export const RegisterScreen = () => {
  return (
    <>
      <h3 className="auth__title mb-5">Register</h3>
      <form>
        <div className="auth__group">
          <input
            type="text"
            className="auth__input"
            name="firstName"
            placeholder="Your  First Name"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>First Name</label>
        </div>
        <div className="auth__group">
          <input
            type="text"
            className="auth__input"
            name="lastName"
            placeholder="Your Last Name"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Last Name</label>
        </div>
        <div className="auth__group">
          <input
            type="email"
            className="auth__input"
            name="email"
            placeholder="Your Email"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Email</label>
        </div>
        <div className="auth__group">
          <input
            type="Password"
            className="auth__input"
            name="password"
            placeholder="Your Password"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Password</label>
        </div>

        <button type="submit" className="btn btn-primary btn-block mb-5">
          Sign In
        </button>

        <Link to="/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </>
  );
};
