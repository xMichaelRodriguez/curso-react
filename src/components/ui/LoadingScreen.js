import React from "react";

export const LoadingScreen = () => {
  return (
    <div>
      <h1>Cargando</h1>
    </div>
  );
};
