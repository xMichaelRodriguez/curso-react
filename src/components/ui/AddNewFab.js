import React from "react";
import { useDispatch } from "react-redux";
import { uiOpenModal } from "../../actions/ui";

export const AddNewFab = () => {
  const dispatch = useDispatch();
  const handleNewEvent = () => {
    dispatch(uiOpenModal());
  };
  return (
    <button className="btn btn-dark fab" onClick={handleNewEvent}>
      <i className="fas fa-plus"></i>
    </button>
  );
};
