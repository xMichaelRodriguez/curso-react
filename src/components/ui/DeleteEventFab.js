import React from "react";
import { useDispatch } from "react-redux";
import { eventDeleted } from "../../actions/eventsCalendar";

export const DeleteEventFab = () => {
  const dispatch = useDispatch();
  const handleDeleted = () => {
    dispatch(eventDeleted());
  };
  return (
    <button className="btn btn-danger fab-denger" onClick={handleDeleted}>
      <i className="fas fa-trash"></i>
      <span> Eliminar Evento</span>
    </button>
  );
};
