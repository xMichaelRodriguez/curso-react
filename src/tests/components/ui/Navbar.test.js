import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent,  render } from '@testing-library/react';
import { MemoryRouter, Router } from 'react-router-dom';
import { Navbar } from '../../../components/ui/Navbar';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';


describe('Test in <Navbar />', () => {
    const historyMock = {
        push: jest.fn(),
        replace:jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),

    }
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: "Michael Rodriguez"
        }
    }
    let component;
    beforeEach(() => {

        component = render(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <Router history={historyMock}>
                        <Navbar />

                    </Router>
                </MemoryRouter>
            </AuthContext.Provider>
        );
    });

    afterEach(() => {
        jest.clearAllMocks();
    });


    test('should display currently', () => {


        expect(component).toMatchSnapshot();
        component.getByText('Michael Rodriguez');
    });

    test('should calls the logout and use history', () => {

        const el = component.getByText('Logout');
        expect(el).toHaveClass('btn')
        fireEvent.click(el);
        expect(contextValue.dispatch).toHaveBeenLastCalledWith({ type: types.logout });
        expect(historyMock.replace).toHaveBeenCalledWith('/login')
    });



});
