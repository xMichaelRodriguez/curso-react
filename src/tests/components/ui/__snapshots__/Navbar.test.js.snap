// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Navbar /> should display currently 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <a
          class="navbar-brand"
        >
          Asociaciones
        </a>
        <div
          class="navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-item nav-link"
            >
              Marvel
            </a>
            <a
              class="nav-item nav-link"
            >
              DC
            </a>
            <a
              class="nav-item nav-link"
            >
              Search
            </a>
          </div>
        </div>
        <div
          class="navbar-collapse collapse w-100 order-3 dual-collapse2"
        >
          <ul
            class="navbar-nav ml-auto"
          >
            <span
              class="nav-item nav-link text-white"
            >
              Michael Rodriguez
            </span>
            <button
              class="nav-item nav-link btn"
            >
              Logout
            </button>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        class="navbar-brand"
      >
        Asociaciones
      </a>
      <div
        class="navbar-collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-item nav-link"
          >
            Marvel
          </a>
          <a
            class="nav-item nav-link"
          >
            DC
          </a>
          <a
            class="nav-item nav-link"
          >
            Search
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse w-100 order-3 dual-collapse2"
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <span
            class="nav-item nav-link text-white"
          >
            Michael Rodriguez
          </span>
          <button
            class="nav-item nav-link btn"
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
